test/testdrive/avro-sources.td

```diff
@@ -226,6 +226,99 @@ a b
+
+$ kafka-create-topic topic=non-dbz-data-multi-partition partitions=2
+
+$ kafka-ingest format=avro topic=non-dbz-data-multi-partition schema=${non-dbz-schema} timestamp=1 partition=1
+{"a": 4, "b": 1}
+
+$ kafka-ingest format=avro topic=non-dbz-data-multi-partition schema=${non-dbz-schema} timestamp=1 partition=0
+{"a": 1, "b": 2}
+
+> CREATE MATERIALIZED SOURCE non_dbz_data_multi_partition
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-non-dbz-data-multi-partition-${testdrive.seed}'
+  WITH (start_offset=0)
+  FORMAT AVRO USING SCHEMA '${non-dbz-schema}'
+  ENVELOPE NONE
+
+> SELECT * FROM non_dbz_data_multi_partition
+a  b
+-----
+1  2
+4  1
+
+> CREATE MATERIALIZED SOURCE non_dbz_data_multi_partition_2
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-non-dbz-data-multi-partition-${testdrive.seed}'
+  WITH (start_offset=[0,0])
+  FORMAT AVRO USING SCHEMA '${non-dbz-schema}'
+  ENVELOPE NONE
+
+> SELECT * FROM non_dbz_data_multi_partition_2
+a  b
+-----
+1  2
+4  1
+
+> CREATE MATERIALIZED SOURCE non_dbz_data_multi_partition_fast_forwarded
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-non-dbz-data-multi-partition-${testdrive.seed}'
+  WITH (start_offset=[0,1])
+  FORMAT AVRO USING SCHEMA '${non-dbz-schema}'
+  ENVELOPE NONE
+
+> SELECT * FROM non_dbz_data_multi_partition_fast_forwarded
+a  b
+----
+1  2
+
+> CREATE MATERIALIZED SOURCE non_dbz_data_multi_partition_fast_forwarded_2
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-non-dbz-data-multi-partition-${testdrive.seed}'
+  WITH (start_offset=[1,0])
+  FORMAT AVRO USING SCHEMA '${non-dbz-schema}'
+  ENVELOPE NONE
+
+> SELECT * FROM non_dbz_data_multi_partition_fast_forwarded_2
+a  b
+----
+4  1
+
+# Test an Avro source without a Debezium envelope with specified offsets and varying partition numbers.
+
+$ kafka-create-topic topic=non-dbz-data-varying-partition partitions=1
+
+$ kafka-ingest format=avro topic=non-dbz-data-varying-partition schema=${non-dbz-schema} timestamp=1 partition=0
+{"a": 5, "b": 6}
+
+# Erroneously adds start_offsets for non-existent partitions.
+> CREATE MATERIALIZED SOURCE non_dbz_data_varying_partition
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-non-dbz-data-varying-partition-${testdrive.seed}'
+  WITH (start_offset=[0,1])
+  FORMAT AVRO USING SCHEMA '${non-dbz-schema}'
+  ENVELOPE NONE
+
+> SELECT * FROM non_dbz_data_varying_partition
+a  b
+----
+5  6
+
+$ kafka-add-partitions topic=non-dbz-data-varying-partition total-partitions=2
+
+# Reading data that's ingested to a new partition takes longer than the default timeout.
+$ set-sql-timeout duration=60s
+
+$ kafka-ingest format=avro topic=non-dbz-data-varying-partition schema=${non-dbz-schema} timestamp=1 partition=1
+{"a": 7, "b": 8}
+{"a": 9, "b": 10}
+
+# Because the start offsets erronously included an offset for partition 1 (which didn't exist at the time),
```

Comment:
This test case is good! What about the opposite cases - growing partitions past the start_offset length (as opposed to growing partitions up to the start_offset length)

Two test cases that I'm curious about:
* if you do start_offset=1, partitions = 1, then grow partitions to 2: will the second partition have a start offset of 1 or 0 (related to previous comment)
* if you do start_offset=[1, 1], partitions = 2, then grow partitions to 3: will the third partition have a start offset of 0

Reply:
In all cases new partitions without a specified offset will start at 0! I've added the tests you've suggested to illustrate that.

Reply:
Thank you!